scripts/python/iscsi_config.py

```diff
@@ -0,0 +1,286 @@
+
+    commands = [
+        # Create target directory
+        "mkdir -p /etc/target",
+        # Mount configfs
+        "mount -t configfs configfs /sys/kernel/config 2>/dev/null || true",
+        # Load kernel modules
+        "modprobe target_core_mod",
+        "modprobe iscsi_target_mod",
+        # Start and enable target service
+        "systemctl start target",
+        "systemctl enable target",
+        # Verify setup
+        "systemctl status target --no-pager",
+    ]
+
+    for cmd in commands:
+        log.info(f"Executing on target VM: {cmd}")
+        success, stdout, stderr = ssh_run(target_vm_ip, cmd, username)
+        if not success and "already" not in stderr.lower():
+            log.warning(f"Command warning: {stderr}")
+        if stdout:
+            log.debug(f"Output: {stdout.strip()}")
+
+
+def get_worker_node_ips(kubeconfig_path):
+    """
+    Collects worker node IPs from the Kubernetes cluster using oc command.
+
+    Parameters:
+    kubeconfig_path (str): Path to the kubeconfig file.
+
+    Returns:
+    list: A list of IP addresses of worker nodes or an empty list on failure.
+    """
+    # Command to get node details in JSON
+    cmd = [
+        "oc",
+        "--kubeconfig",
+        kubeconfig_path,
+        "get",
+        "nodes",
+        "-l",
+        "node-role.kubernetes.io/worker",
+        "-o",
+        "json",
+    ]
+
+    try:
+        result = subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError as e:
+        log.error("Error running oc command:", e.output.decode())
+        return []
+
+    data = json.loads(result)
+
+    worker_ips = []
+
+    # Parse each worker node
+    for node in data.get("items", []):
+        for addr in node["status"]["addresses"]:
+            if addr["type"] == "InternalIP":
+                worker_ips.append(addr["address"])
+
+    return worker_ips
+
+
+# --------------------------------------------------------
+# STEP 1: Collect worker node initiator IQNs
+# --------------------------------------------------------
+def get_worker_iqns(worker_node_ips):
+    """
+    Configures the iSCSI target with given IQNs and IP.
+    This function is a placeholder and requires implementation.
+
+    Parameters:
+    target_iqn (str): The IQN of the target node.
+    target_ip (str): The IP address of the target node.
+    worker_iqns (list): List of IQNs of worker nodes.
+    """
+    iqns = []
+    ("\n=== Collecting Worker IQNs ===")
+    for node_ip in worker_node_ips:
+        check_iscsi_command = "which iscsiadm"
+        iscsiadm_installed = ssh_run(node_ip, check_iscsi_command, username="core")
+        if not iscsiadm_installed:
+            log.info(f"[{node_ip}] iscsiadm not found. Installing...")
+            ssh_run(
+                node_ip,
+                "sudo yum install -y iscsi-initiator-utils || sudo apt install -y open-iscsi",
+                username="core",
+            )
```

Comment:
it will be installed by default during the OCP cluster deployment. It is not required